src/main/java/com/marklogic/spark/reader/file/ArchiveFileReader.java

```diff
@@ -12,17 +13,20 @@
 import org.apache.spark.sql.catalyst.InternalRow;
 import org.apache.spark.sql.connector.read.PartitionReader;
 
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.ObjectOutputStream;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipInputStream;
 
-class ArchiveFileReader implements PartitionReader<InternalRow> {
+public class ArchiveFileReader implements PartitionReader<InternalRow> {
```

Comment:
I'm wondering if this should be two classes. One for streaming and one for not streaming. Is that even possible?

Reply:
Definitely possible. It's a choice between duplicating the workflow in another class - i.e. there's a nontrivial amount for iterating over multiple entries and multiple riles - and trying to support both streaming and non-streaming. I'll keep it in mind though - it would be a refactoring supported by the tests in place.